src/angular/ngWalker.ts

```diff
@@ -107,56 +118,48 @@ export class NgWalker extends Lint.RuleWalker {
     let name = getDecoratorName(decorator);
 
     if (name === 'Injectable') {
-      this.visitNgInjectable(<ts.ClassDeclaration>decorator.parent, decorator);
+      this.visitNgInjectable(decorator.parent as ts.ClassDeclaration, decorator);
     }
 
     if (name === 'Pipe') {
-      this.visitNgPipe(<ts.ClassDeclaration>decorator.parent, decorator);
+      this.visitNgPipe(decorator.parent as ts.ClassDeclaration, decorator);
     }
 
     if (name === 'NgModule') {
       this.visitNgModule(decorator);
     }
-
-    // Not invoked @Component or @Pipe, or @Directive
-    if (
-      !(<ts.CallExpression>decorator.expression).arguments ||
-      !(<ts.CallExpression>decorator.expression).arguments.length ||
-      !(<ts.ObjectLiteralExpression>(<ts.CallExpression>decorator.expression).arguments[0]).properties
-    ) {
-      return;
-    }
   }
 
   protected visitNgComponent(metadata: ComponentMetadata) {
-    const template = metadata.template;
-    const getPosition = (node: ts.Node) => {
-      let pos = 0;
-      if (node) {
-        pos = node.pos + 1;
-        try {
-          pos = node.getStart() + 1;
-        } catch (e) {}
-      }
-      return pos;
-    };
-
     const { styles = [] } = metadata;
 
     for (const style of styles) {
       try {
-        this.visitNgStyleHelper(parseCss(style.style.code), metadata, style, getPosition(style.node!));
+        const cssAst = parseCss(style.style.code);
+        this.visitNgStyleHelper(cssAst, metadata, style, getPosition(style.node!));
       } catch (e) {
-        logger.error('Cannot parse the styles of', ((<any>metadata.controller || {}).name || {}).text, e);
+        const {
+          controller: { name }
+        } = metadata;
```

Comment:
This needs defaults in order to guard the same way the original did. Not sure if intentional.

Reply:
The _property_ `controller` always exists according to:

https://github.com/mgechev/codelyzer/blob/da0f553fe7d580f38907f9baa9124cbc55e37c96/src/angular/metadata.ts#L23-L27

... and it makes sense, because once we instantiate a `DirectiveMetadata`, we pass all properties:

https://github.com/mgechev/codelyzer/blob/da0f553fe7d580f38907f9baa9124cbc55e37c96/src/angular/metadataReader.ts#L63-L67

... now I'm thinking: why do we need to use `Object.assign` instead of a trivial `constructor`?

<hr>

Anyway, these "defaults" should have been changed in #631, but I really forgot about it. :disappointed: